### PR TITLE
[NO-ISSUE] fix: check service props has passed in list table block

### PR DESCRIPTION
**Pull request by lucasmendes21**
### Explain what was fixed and the correct behavior.
check service props has passed in list table block

### Does this PR introduce UI changes? Add a video or screenshots here.
https://jam.dev/c/1c6d91f1-2f7e-4644-a825-06725a151211

<hr />

#### Make sure your pull request fits the checklist below (when applicable):

- [x] The issue title follows the format: [ISSUE_CODE] bug: TITLE
- [x] Commits are tagged with the right word (feat, test, refactor, etc)
- [ ] Application responsiveness was tested to different screen sizes
- [ ] New tests are added to prevent the same issue from happening again
- [ ] UI changes are validated by a team designer
- [x] Code is formatted and linted
- [x] Tags are added to the PR

#### These changes were tested on the following browsers:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox
- [ ] Safari
- [x] Brave

### CODE REVIEW

File: src/templates/list-table-block/index.vue [hunk 1]
```diff
@@ -391,12 +391,14 @@
...
-        : await props.listService({ page })
-      data.value = response
+      let response
+      if (props.listService) {
+        isLoading.value = true
+        response = props.isGraphql ? await props.listService() : await props.listService({ page })
+        data.value = response
+      } else {
+        data.value = []
+      }
```
Comment (HerbertJulio) [inline src/templates/list-table-block/index.vue:401]: Sugestão:
```suggestion
      data.value = []
      if (props.listService) {
        isLoading.value = true
        response = props.isGraphql ? await props.listService() : await props.listService({ page })
        data.value = response
      } 
```

Comment (HerbertJulio) [inline src/templates/list-table-block/index.vue:401]:
```suggestion
      data.value = [];
      if (props.listService) {
        try {
          isLoading.value = true;
          const response = props.isGraphql 
            ? await props.listService() 
            : await props.listService({ page });
          data.value = response;
        } catch (error) {
          console.error("Error fetching data:", error);
        } finally {
          isLoading.value = false;
        }
      }
```

Comment (matheus-sunderhus) [reply src/templates/list-table-block/index.vue:401]: console.error("Error fetching data:", error); recomendo remover essa parte e colocar o toast. De resto acredito que o refactor do @HerbertJulio  controla melhor o estado do listService.

File: src/views/WafRules/ListWafRulesTuning.vue [hunk 2]
```diff
@@ -49,6 +49,7 @@
     :editInDrawer="openMoreDetails"
     emptyListMessage="No requests found."
     isTabs
+    :actions="[]"
```
Comment (matheus-sunderhus) [inline src/views/WafRules/ListWafRulesTuning.vue:52]: transformar o [] em uma ref, 
const noActionToDisplay = ref([]).

dessa forma passamos melhor o contexto do porque está sendo passado o array vazio para essa prop.

**Review by aloisio-m-bastian — APPROVED** [review]

**Review by HerbertJulio — APPROVED** [review]

**Review by aloisio-m-bastian — APPROVED** [review]

**Pull request merged**